examples/rando/src/rando.cc

```diff
@@ -21,24 +19,55 @@ limitations under the License.
 
 #include "rando.h"
 
-using namespace Plugin;
+using Plugin::Config;
+using Plugin::ConfigPolicy;
+using Plugin::Metric;
+using Plugin::Meta;
+using Plugin::Type;
 
-Config::Policy Rando::getConfigPolicy() {
-  Config::Policy policy;
+ConfigPolicy Rando::get_config_policy() {
+  ConfigPolicy policy;
   return policy;
 }
 
-std::vector<Metric::Metric> Rando::getMetricTypes(Config::Config cfg) {
-  std::vector<Metric::Metric> metrics;
+std::vector<Metric> Rando::get_metric_types(Config cfg) {
+  std::vector<Metric> metrics;
+  metrics.push_back({
+    { // namespace initializer_list
+      {"intel", "", ""},
+      {"cpp", "", ""},
+      {"mock", "", ""},
+      {"random_number", "", ""},
+      {"one", "", ""},
+    },
+    "", // unit
+    "the first random number" // description
+  });
+  metrics.push_back({
+    { // namespace initializer_list
+      {"intel", "", ""},
+      {"cpp", "", ""},
+      {"mock", "", ""},
+      {"random_number", "", ""},
+      {"two", "", ""},
+    },
+    "", // unit
+    "the second random number" // description
+  });
   return metrics;
 }
 
-std::vector<Metric::Metric> Rando::collectMetrics(std::vector<Metric::Metric> metrics) {
-  return metrics;
+void Rando::collect_metrics(std::vector<Metric>* metrics) {
```

Comment:
Instead of a pointer, have you considered to use a smart pointer instead?http://en.cppreference.com/w/cpp/memory/shared_ptr